src/webots/nodes/WbPositionSensor.cpp

```diff
@@ -56,8 +56,11 @@ void WbPositionSensor::init() {
   mResolution = findSFDouble("resolution");
 }
 
+void WbPositionSensor::preFinalize() {
+  WbJointDevice::preFinalize();
+}
+
 void WbPositionSensor::postFinalize() {
-  WbJointDevice::postFinalize();
```

Comment:
Independently from the other changes, this should be kept in any case because it is important to call the base class method: recursively it should always call `WbBaseNode::postFinalize` to work properly.